File: python/palindrome.py
def palindrome(word):
    if type(word) == str:
        word = word.lower()
    if type(word) == int:
        word = str(word)
    punctuations = '''!() -[]{};:'"\,<>./?@#$%^&*_~'''
    cleaned = ""
    for char in word:
        if char not in punctuations:
            cleaned = cleaned + char
    half_len = len(cleaned) // 2
    for i in range(0, half_len):
        if cleaned[i] != cleaned[len(cleaned) - 1 - i]:
            return False
    return True

File: js/palindrome.js
function palindrome(word) {
  if (typeof word == "string") {
    word = word.toLowerCase();
  }
  if (typeof word == "number") {
    word = word.toString();
  }
  let cleaned = word.replace(/[.,\/#!$% \^&\*;:{}=\-_`~()]/g,"");
  let half_len = Math.floor(cleaned.length / 2);
  for (let i = 0; i < half_len; ++i) {
    if (cleaned[i] != cleaned[cleaned.length - 1 - i]) {
      return false;
    }
  }
  return true;
};

module.exports = palindrome;